icaruscode/Timing/PMTTimingCorrectionsProvider.cxx

```diff
@@ -10,114 +10,107 @@
 #include "messagefacility/MessageLogger/MessageLogger.h"
 #include "cetlib_except/exception.h"
 
-#include "lardata/DetectorInfoServices/DetectorClocksService.h"
-
+// Local
 #include "icaruscode/Timing/PMTTimingCorrections.h"
 #include "icaruscode/Timing/PMTTimingCorrectionsProvider.h"
 
 // Database interface helpers
-#include "wda.h"
+#include "larevt/CalibrationDBI/Providers/DBFolder.h"
 
 // C/C++ standard libraries
-#include <memory> // std::unique_ptr<>
-#include <optional>
 #include <string>
-#include <utility> // std::move()
-#include <cassert>
-#include <tuple>
+#include <vector>
 
 //--------------------------------------------------------------------------------
 
 icarusDB::PMTTimingCorrectionsProvider::PMTTimingCorrectionsProvider
     (const fhicl::ParameterSet& pset) 
-    : fUrl{ pset.get<std::string>("DatabaseURL", "https://dbdata0vm.fnal.gov:9443/icarus_con_prod/app/data?") }
-    , fTimeout{ pset.get<unsigned int>("Timeout", 1000) }
-    , fVerbose{ pset.get<bool>("Verbose", false) }
+    : fVerbose{ pset.get<bool>("Verbose", false) }
     , fLogCategory{ pset.get<std::string>("LogCategory", "PMTTimingCorrection") }
-    , fClocksData{ art::ServiceHandle<detinfo::DetectorClocksService const>()->DataForJob() }
-    { }
-
+    , fTags{ pset.get<fhicl::ParameterSet>("CorrectionsTags") }
+    { 
+	fCablesTag  = fTags.get<std::string>("CablesTag", "v1r0");
+	fLaserTag   = fTags.get<std::string>("LaserTag", "v1r0");
+	fCosmicsTag = fTags.get<std::string>("CosmicsTag", "v1r0");
```

Comment:
The `CorrectionsTags` (shouldn't it be `CorrectionTags` in English?) table does not need to be stored, since its information is fully extracted into other data members:
```suggestion
    { 
        fhicl::ParameterSet const tags{ pset.get<fhicl::ParameterSet>("CorrectionsTags") };
	fCablesTag  = tags.get<std::string>("CablesTag", "v1r0");
	fLaserTag   = tags.get<std::string>("LaserTag", "v1r0");
	fCosmicsTag = tags.get<std::string>("CosmicsTag", "v1r0");
```
Chances are that you can even refer to the parameters directly:
```suggestion
    { 
	fCablesTag  = tags.get<std::string>("CorrectionsTags.CablesTag", "v1r0");
	fLaserTag   = tags.get<std::string>("CorrectionsTags.LaserTag", "v1r0");
	fCosmicsTag = tags.get<std::string>("CorrectionsTags.CosmicsTag", "v1r0");
```
Some of `fhicl::ParameterSet` methods support this, I don't remember in which circumstances though.

Reply:
Thank you for the suggestions. I was unsure on how to read tags from a table, so I went and saved the table as well. I will simplify the code as you suggests.